### PR TITLE
Fix sudo issue

### PR DESCRIPTION
**Pull request by rewqazxv**
Fix sudo issue on ArchLinux, which will set the environment variable 'SUDO_COMMAND' to '/usr/bin/su' or '/usr/bin/zsh'

### CODE REVIEW

**Comment by exitmsconfig** [comment]
该用户作者，邮箱gzhangz92@gmail.com收取用户费用后不处理问题，请见证

File: acme.sh [hunk 1]
```diff
@@ -6362,11 +6362,18 @@ _checkSudo() {
...
+        ;;
+    esac
+    for i in $(cat /etc/shells); do
+      if [ "$SUDO_COMMAND" = "$i" ]; then
+        #it's a normal user running `sudo -i` or `sudo -s`, fine
+        return 0
+      fi
+    done
     #otherwise
     return 1
```
Comment (honzahommer) [inline acme.sh]:
```
grep "^$SUDO_COMMAND\$" /etc/shells >/dev/null 2>&1
return $?
```

File: acme.sh [hunk 2]
```diff
@@ -6362,11 +6362,18 @@ _checkSudo() {
...
+        #it's a normal user doing `sudo su`, no problem
+        return 0
+        ;;
+    esac
+    for i in $(cat /etc/shells); do
+      if [ "$SUDO_COMMAND" = "$i" ]; then
+        #it's a normal user running `sudo -i` or `sudo -s`, fine
+        return 0
+      fi
+    done
```
Comment (honzahommer) [inline acme.sh]: Just suggestion...
```sh
if [ -n "$SUDO_COMMAND" ]; then
   #it's a normal user doing "sudo su", or `sudo -i` or `sudo -s`
  _endswith "$SUDO_COMMAND" /bin/su || grep "^$SUDO_COMMAND\$" /etc/shells >/dev/null 2>&1
  return $?
fi
```

Comment (rewqazxv) [reply acme.sh]: Very concise. Thank you.

**Comment by neirbowj** [comment]
What is blocking progress for this PR?

**Comment by dlangille** [comment]
FYI, this was just patched into the FreeBSD port: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=248434

**Comment by Neilpang** [comment]
@dlangille   Thanks.

**Pull request merged**